Paintroid/src/main/java/org/catrobat/paintroid/ui/MainActivityNavigator.kt

```diff
@@ -328,15 +327,16 @@ class MainActivityNavigator(
     }
 
     override fun showIndeterminateProgressDialog() {
-        if (progressDialog == null) {
-            progressDialog = newInstance(mainActivity)
-        }
-        progressDialog?.show()
+        val progressDialogFragment = IndeterminateProgressDialog()
+        showDialogFragmentSafely(progressDialogFragment, INDETERMINATE_PROGRESS_DIALOG_TAG)
     }
 
     override fun dismissIndeterminateProgressDialog() {
-        progressDialog?.dismiss()
-        progressDialog = null
+        val progressDialogFragment = findFragmentByTag(INDETERMINATE_PROGRESS_DIALOG_TAG)
+        progressDialogFragment?.let {
+            progressDialogFragment as DialogFragment
+            progressDialogFragment.dismiss()
+        }
```

Comment:
```suggestion
        val progressDialogFragment = findFragmentByTag(INDETERMINATE_PROGRESS_DIALOG_TAG) as DialogFragment?
        progressDialogFragment?.dismiss()
```

A slight improvement, also reduces some number of line ;D

Reply:
ah yes thats a nice way ^^